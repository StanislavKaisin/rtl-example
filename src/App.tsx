import React, { useState } from "react";
import logo from "./logo.svg";
import "./App.css";

interface ISearch {
  value: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  children: any;
}

const Search = ({ value, onChange, children }: ISearch) => {
  return (
    <div>
      <label htmlFor="search">{children}</label>
      <input id="search" type="text" value={value} onChange={onChange} />
    </div>
  );
};

function App() {
  const [search, setSearch] = useState("");
  const handleChange = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
    setSearch(target.value);
  };
  return (
    <div className="App">
      <Search value={search} onChange={handleChange}>
        Search:
      </Search>
      <p>Searches for {search ? search : "..."}</p>
    </div>
  );
}

export default App;
