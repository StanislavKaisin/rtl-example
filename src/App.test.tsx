import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";

test("renders a component", () => {
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
  // screen.debug();
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});
