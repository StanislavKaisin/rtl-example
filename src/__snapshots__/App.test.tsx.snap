// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <label
        for="search"
      >
        Search:
      </label>
      <input
        id="search"
        type="text"
        value=""
      />
    </div>
    <p>
      Searches for ...
    </p>
  </div>
</DocumentFragment>
`;
